### PR TITLE
Fix #1257 and some lint

### LINKED ISSUE #1257: useComboBox: document is not defined getStatusDiv React Native
- `downshift` version: ^6.1.2
- `expo`: ^40.0.0,

**Relevant code or config**

```
import color from 'color';
import { useCombobox } from 'downshift';
import React, { useLayoutEffect, useRef, useState } from 'react';
import {
  FlatList,
  Pressable,
  StyleSheet,
  TouchableWithoutFeedback,
  useWindowDimensions,
  View
} from 'react-native';
import {
  Portal,
  Surface,
  Text,
  TextInput,
  TouchableRipple,
  useTheme
} from 'react-native-paper';

function ComboBox({
  containerProps = {},
  dropdownProps = {},
  options = [],
  onSelect = option => option,
  getOptionLabel = option => <Text>{option}</Text>,
  optionStyles = {},
  containerStyle = {},
  inputValue: inputValueProp,
  value,
  onInputValueChange = () => {},
  openOnOptions = false,
  itemToString = item => (item ? item.toString() : null),
  ...props
}) {
  const {
    closeMenu,
    getComboboxProps,
    getInputProps,
    getItemProps,
    highlightedIndex,
    isOpen,
    openMenu,
    toggleMenu,
    selectItem,
    selectedItem,
    inputValue
  } = useCombobox({
    items: options,
    inputValue: inputValueProp,
    itemToString,
    selectedItem: value,
    onSelectedItemChange: ({ selectedItem }) => onSelect(selectedItem),
    onInputValueChange
  });

  const window = useWindowDimensions();
  const theme = useTheme();
  const [layout, setLayout] = useState();
  const inputRef = useRef(null);

  const onLayout = (x, y, width, height, pagex, pagey) => {
    setLayout({ x, y, width, height, pagex, pagey });
  };

  const titleColor = color(theme.colors.text).alpha(0.87).rgb().string();
  const descriptionColor = color(theme.colors.text).alpha(0.54).rgb().string();

  const computeDropdownPosition = () => {
    let maxHeight = window.height - (layout.pagey + layout.height - 15);
    if (layout.pagey > (window.height / 3) * 2) {
      maxHeight = window.height - (window.height - layout.pagey + layout.height - 15);
      let bottom = window.height - layout.pagey;
      if (!searchable) {
        bottom -= layout.height + 7;
      }
      return {
        maxHeight,
        bottom,
        left: layout.pagex + 5,
        width: layout.width - 10
      };
    }
    return {
      maxHeight,
      top: layout.pagey + layout.height - 15,
      left: layout.pagex + 5,
      width: layout.width - 10
    };
  };

  useLayoutEffect(() => {
    if (inputRef.current) inputRef.current.measure(onLayout);
  }, [isOpen]);

  const renderItem = ({ item, index }) => {
    let Component = getOptionLabel(item, {
      titleStyles: [styles.title, { color: titleColor }],
      descriptionStyles: [styles.description, { color: descriptionColor }]
    });
    if (typeof Component === 'string') {
      Component = <Text>{Component}</Text>;
    }

    const props = getItemProps({ item, index });
    return (
      <TouchableRipple key={index} style={[styles.optionContainer, optionStyles]} {...props}>
        <View style={[styles.item, styles.content]} highlighted={highlightedIndex === index}>
          {Component}
        </View>
      </TouchableRipple>
    );
  };
  const comboBoxProps = getComboboxProps(undefined, { suppressRefError: true });
  const inputProps = getInputProps(props);
  return (
    <View {...containerProps} {...comboBoxProps} style={[{ position: 'relative' }, containerStyle]}>
      <View ref={inputRef} onLayout={() => inputRef.current.measure(onLayout)}>
        <Pressable onPress={openMenu}>
          <TextInput
            right={
              <TextInput.Icon
                name={isOpen ? 'chevron-up' : 'chevron-down'}
                disabled={options.length === 0 && openOnOptions}
                onPress={toggleMenu}
              />
            }
            {...inputProps}
          />
        </Pressable>
      </View>
      {isOpen && (
        <Portal>
          <TouchableWithoutFeedback onPress={closeMenu}>
            <View style={{ flex: 1 }} />
          </TouchableWithoutFeedback>
          <View
            style={{
              position: 'absolute',
              ...computeDropdownPosition()
            }}
          >
            <Surface style={{ position: 'relative' }}>
              <FlatList
                data={options}
                renderItem={renderItem}
                /*
                 * Keep `disableScrollViewPanResponder` prop, else onPress events will not be fired, because "reasons"
                 * https://github.com/facebook/react-native/issues/20011#issuecomment-565740917
                 */
                disableScrollViewPanResponder
              />
            </Surface>
          </View>
        </Portal>
      )}
    </View>
  );
}
```

**What you did**:
Using `useComboBox`, set up like above, but after inputting into the combobox textfield, I receive the following error:

Screen Shot 2021-03-31 at 2 38 02 PM

Unsure what is going on there.

### PR DESCRIPTION
**Pull request by DaveWelling**
This fixes the bug with rendering in react-native as described in issue document #1257 

Closes https://github.com/downshift-js/downshift/issues/1257

This will allow the react-native version of useCombobox and useSelect to work instead of crashing.

I emulated similar code in in the main downshift component (downshift.js line 1171) by pulling in
`import {isReactNative} from '../is.macro'`
and then avoiding the `updateA11yStatus` method (which manipulates the DOM and will not work in react-native).

**Checklist**:

- [ ] Documentation N/A
- [ ] Tests N/A
- [ ] TypeScript Types N/A
- [ ] Flow Types N/A
- [x ] Ready to be merged 
      
Yes

This is a very small change which emulates similar code already existing in this repository (downshift.js line 1171).

### CODE REVIEW

**Comment by silviuaavram** [comment]
Awesome work @DaveWelling ! Thanks for fixing this! I will merge once CI passes.

**Comment by silviuaavram** [comment]
Something seems to be broken in our CI. We should take a look at it.

**Comment by silviuaavram** [comment]
@DaveWelling any idea what that import is all about? I took a look at your changes and it's not clear why they caused it. Thanks!

**Comment by DaveWelling** [comment]
Are you talking about 
```
// eslint-disable-next-line import/extensions
const config = require('kcd-scripts/dist/config/rollup.config.js')
```
?
I think that is because leaving off the .js will result in parsers thinking the file is `rollup.config`.

**Comment by DaveWelling** [comment]
Oh! you are talking about this:
[lint] Error: Error while loading rule 'import/no-import-module-exports': Cannot find module '/home/runner/work/downshift/downshift/dist/downshift.cjs.js'. Please verify that the package.json has a valid "main" entry
[lint] Occurred while linting /home/runner/work/downshift/downshift/cypress/integration/combobox.js
I agree, it seems as though there is nothing in my changes that would cause that.  Are other changes merging ok now?

**Comment by DaveWelling** [comment]
@silviuaavram Is there a way to bypass this validation?  All linting passes locally.  This seems to be something to do with cypress and I've made no changes that should affect that.  
When I attempt to run cypress locally it fails however because of docz and that project appears to be rotting (https://www.bountysource.com/issues/98938185-sorry-everybody-i-failed-with-you).  
I tracked the first error down to node_modules\get-pkg-repo\src\index.js 
This error is fixed by doing this:
```
//let result = hostedGitInfo.fromUrl(repositoryURL) || parseRepositoryURL(repositoryURL);
  let result = parseRepositoryURL(repositoryURL);
```
However, when I fix that, it fails again in Docz with this:
```
 ERROR #85923  GRAPHQL

There was an error in your GraphQL query:

Cannot query field "allDoczEntries" on type "Query".
```
This is in File: ..\node_modules\gatsby-theme-docz\lib\createPagesStatefully.js:28:10
which is used by docz:
```
npm ls gatsby-theme-docz
downshift@0.0.0-semantically-released D:\Temp\junk\downshift
`-- docz@2.3.1
  `-- gatsby-theme-docz@2.3.1
```

Long story short:  I don't think it makes sense to prevent fixes because there is a problem with a completely unrelate section of the project which has a dependency on a rotting project.  What do you think?
